PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/DetectDiscouragedAPIUsageCheck.java

```diff
@@ -95,16 +93,28 @@ public void visitElement(@NotNull PsiElement element) {
                 // resolvedMethod is the method that is being called
                 PsiElement resolvedMethod = methodExpression.resolve();
 
-                // check if the method is a connectionString call
-                if (resolvedMethod != null && resolvedMethod instanceof PsiMethod && ((PsiMethod) resolvedMethod).getName().equals(METHOD_TO_CHECK)) {
+                // check if the method is a discouraged API call by accessing the keys of the map stored in the configuration file
+                if (resolvedMethod != null && resolvedMethod instanceof PsiMethod
+                        && METHODS_TO_CHECK.containsKey(((PsiMethod) resolvedMethod).getName())) {
+
                     PsiMethod method = (PsiMethod) resolvedMethod;
 
+                    // containingClass is the client class that is being called. check if the class is an azure
+
                     // containingClass is the client class that is being called. check if the class is an azure client
                     PsiClass containingClass = method.getContainingClass();
 
                     // compare the package name of the containing class to the azure package name from the configuration file
-                    if (containingClass != null && containingClass.getQualifiedName() != null && containingClass.getQualifiedName().startsWith(RuleConfig.AZURE_PACKAGE_NAME)) {
-                        holder.registerProblem(element, SUGGESTION);
+                    if (containingClass != null && containingClass.getQualifiedName() != null
```

Comment:
We don't check the getCompletions is called from Open AI context?

Reply:
added a "com.azure.ai.openai" magic string